### PR TITLE
Sp2 front end

### PR DESCRIPTION
**Pull request by tkiethanom**
r? @rlr @tspurway 

Init API end point now being used to fetch Channels and Countries. Added front end Validation for Account and Campaign creation and updating. Added date picker and fixed date formatting when sending to API.

### CODE REVIEW

**Comment by rlr** [comment]
oops. my bad. I'll go through it today

File: front_end/src/actions/Accounts/AccountActions.js [hunk 1]
```diff
@@ -75,13 +75,8 @@ export function createAccount(data) {
...
       .then(response => response.json())
-      .then((json) => new Promise(resolve => {
-        if(json.result !== undefined){
-          dispatch(receiveCreateAccount(json.result));
-        }
-        else{
-          dispatch(receiveCreateAccount(null));
-        }
+      .then(json => new Promise(resolve => {
+        dispatch(receiveCreateAccount(json));
```
Comment (rlr) [inline front_end/src/actions/Accounts/AccountActions.js]: :+1: liking all this cleanup

File: front_end/src/actions/Init/InitActions.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      .then(json => new Promise(resolve => {
+        dispatch(receiveInit(json.result));
+        resolve();
+      })
+    );
+  };
+}
+
+export function getChannel(id, rows){
+  return _.find(rows, function(row) {
```
Comment (rlr) [inline front_end/src/actions/Init/InitActions.js]: Any reason to use this over `rows.find(row => row === id)`?

File: front_end/src/components/Accounts/AccountDetails/AccountDetails.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+.details-panel.account-details{
+  /*strong{
+    color: $color-gray;
+    font-weight: 600;
+  }
+
+  .account-details-top{
+    padding: 15px;
+    font-weight: 600;
+  }*/
```
Comment (rlr) [inline front_end/src/components/Accounts/AccountDetails/AccountDetails.scss]: nothing here. Is this placeholder for something coming or should the file be removed?

Comment (tkiethanom) [inline front_end/src/components/Accounts/AccountDetails/AccountDetails.scss]: In the future the Account Details will have a specific style.

File: front_end/src/actions/Init/InitActions.js (hunk 2 above)
Comment (tkiethanom) [inline front_end/src/actions/Init/InitActions.js]: Fixed this for you in the latest push.

**Comment by rlr** [comment]
I think this is good to merge. r+ from me :+1:

**Comment by tspurway** [comment]
go ahead and merge/close this PR @rlr

**Pull request merged**